<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EntityEditorDialog</class>
 <widget class="QDialog" name="EntityEditorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>754</width>
    <height>741</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Game Object Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_9">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QGroupBox" name="mPreviewGB">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>300</width>
         <height>300</height>
        </size>
       </property>
       <property name="title">
        <string>Preview</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="Preview" name="mPreview" native="true">
          <property name="minimumSize">
           <size>
            <width>300</width>
            <height>300</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="mLightingGB">
          <property name="title">
           <string>Lighting</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <layout class="QFormLayout" name="formLayout_5">
               <property name="fieldGrowthPolicy">
                <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="mDiffuseLbl">
                 <property name="text">
                  <string>Diffuse</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSlider" name="mDiffuseSldr">
                 <property name="minimumSize">
                  <size>
                   <width>120</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="maximum">
                  <number>100</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="tickPosition">
                  <enum>QSlider::NoTicks</enum>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="mAmbientLbl">
                 <property name="text">
                  <string>Ambient</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSlider" name="mAmbientSldr">
                 <property name="maximum">
                  <number>100</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="invertedAppearance">
                  <bool>false</bool>
                 </property>
                 <property name="invertedControls">
                  <bool>false</bool>
                 </property>
                 <property name="tickPosition">
                  <enum>QSlider::NoTicks</enum>
                 </property>
                 <property name="tickInterval">
                  <number>0</number>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <layout class="QVBoxLayout" name="verticalLayout_8"/>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="mColorLbl">
                 <property name="text">
                  <string>Light Color</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <layout class="QHBoxLayout" name="horizontalLayout_4">
                 <item>
                  <spacer name="horizontalSpacer">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="ClickableFrame" name="mColorFrame">
                   <property name="minimumSize">
                    <size>
                     <width>40</width>
                     <height>40</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>40</width>
                     <height>40</height>
                    </size>
                   </property>
                   <property name="frameShape">
                    <enum>QFrame::Box</enum>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Sunken</enum>
                   </property>
                   <layout class="QVBoxLayout" name="verticalLayout_7"/>
                  </widget>
                 </item>
                </layout>
               </item>
               <item row="4" column="1">
                <spacer name="verticalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>51</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QPushButton" name="mIconGenBtn">
               <property name="text">
                <string>Generate Icon</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="mPrevDefaultBtn">
               <property name="text">
                <string>Default</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QGroupBox" name="mIconGB">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>85</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="title">
              <string>Icon</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QLabel" name="mIconLbl">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>48</width>
                  <height>48</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>50</height>
                 </size>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="alignment">
                 <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QFormLayout" name="formLayout">
           <property name="sizeConstraint">
            <enum>QLayout::SetDefaultConstraint</enum>
           </property>
           <property name="fieldGrowthPolicy">
            <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="mNameLbl">
             <property name="text">
              <string>Name</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="mNameLE">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="mFilterLbl">
             <property name="text">
              <string>Filter</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <item>
              <widget class="QComboBox" name="mFilterCB">
               <item>
                <property name="text">
                 <string>None</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="mFilterBtn">
               <property name="maximumSize">
                <size>
                 <width>30</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">QPushButton {
min-width: 30px;
}</string>
               </property>
               <property name="text">
                <string>New</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="2" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLineEdit" name="mIconLE"/>
             </item>
             <item>
              <widget class="QPushButton" name="mIconBtn">
               <property name="maximumSize">
                <size>
                 <width>30</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">QPushButton {
min-width: 30px;
}</string>
               </property>
               <property name="text">
                <string>...</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Icon Path</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="mCompGB">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Components</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
         <property name="checkable">
          <bool>false</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QListWidget" name="mCompLW">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QToolButton" name="mDelCompBtn">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../toolkit.qrc">
                <normaloff>:/VecTB/icons/toolbars/Delete.png</normaloff>
                <disabledoff>:/VecTB/icons/toolbars/Delete_Disabled.png</disabledoff>:/VecTB/icons/toolbars/Delete.png</iconset>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="mAddCompBtn">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../toolkit.qrc">
                <normaloff>:/VecTB/icons/toolbars/New_Map.png</normaloff>
                <disabledoff>:/VecTB/icons/toolbars/New_Map_Disabled.png</disabledoff>:/VecTB/icons/toolbars/New_Map.png</iconset>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QStackedWidget" name="mCompW">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>200</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <widget class="QWidget" name="mCompWPage1"/>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="mButtonGroupLayout">
     <item>
      <spacer name="horizontalSpacer_7">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="mOkayBtn">
       <property name="text">
        <string>&amp;Okay</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="mCancelBtn">
       <property name="text">
        <string>&amp;Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Preview</class>
   <extends>QWidget</extends>
   <header>preview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ClickableFrame</class>
   <extends>QFrame</extends>
   <header>clickableframe.h</header>
   <container>1</container>
   <slots>
    <signal>clicked()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../toolkit.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>mIconGenBtn</sender>
   <signal>pressed()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onGenerateIcon()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>678</y>
    </hint>
    <hint type="destinationlabel">
     <x>50</x>
     <y>614</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mPrevDefaultBtn</sender>
   <signal>pressed()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onPrevDefault()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>678</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>618</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mDiffuseSldr</sender>
   <signal>valueChanged(int)</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onDiffuseChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>190</x>
     <y>382</y>
    </hint>
    <hint type="destinationlabel">
     <x>9</x>
     <y>459</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mAmbientSldr</sender>
   <signal>valueChanged(int)</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onAmbientChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>197</x>
     <y>407</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>528</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mCompLW</sender>
   <signal>itemPressed(QListWidgetItem*)</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onCompItemPressed(QListWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>646</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>455</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mOkayBtn</sender>
   <signal>pressed()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onOkay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>662</x>
     <y>730</y>
    </hint>
    <hint type="destinationlabel">
     <x>688</x>
     <y>632</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mCancelBtn</sender>
   <signal>pressed()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onCancel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>743</x>
     <y>730</y>
    </hint>
    <hint type="destinationlabel">
     <x>753</x>
     <y>555</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mColorFrame</sender>
   <signal>clicked()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onColorChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>350</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>622</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mNameLE</sender>
   <signal>editingFinished()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onNameChange()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>596</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>570</x>
     <y>32</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mIconBtn</sender>
   <signal>pressed()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onChooseIcon()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>739</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>753</x>
     <y>102</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mDelCompBtn</sender>
   <signal>pressed()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onRemoveComp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>688</x>
     <y>338</y>
    </hint>
    <hint type="destinationlabel">
     <x>280</x>
     <y>740</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mAddCompBtn</sender>
   <signal>pressed()</signal>
   <receiver>EntityEditorDialog</receiver>
   <slot>onAddComp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>711</x>
     <y>339</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>740</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onGenerateIcon()</slot>
  <slot>onPrevDefault()</slot>
  <slot>onAddComp()</slot>
  <slot>onRemoveComp()</slot>
  <slot>onOkay()</slot>
  <slot>onCancel()</slot>
  <slot>onNewFilter()</slot>
  <slot>onIconBrowse()</slot>
  <slot>onAmbientChange(int)</slot>
  <slot>onDiffuseChange(int)</slot>
  <slot>onCompItemPressed(QListWidgetItem*)</slot>
  <slot>onColorChange()</slot>
  <slot>onNameChange()</slot>
  <slot>onChooseIcon()</slot>
 </slots>
</ui>
